clients/admin-ui/src/features/datastore-connections/system_portal_config/ConnectionListDropdown.tsx

```diff
@@ -167,7 +169,8 @@ const ConnectionListDropdown: React.FC<SelectDropdownProps> = ({
   const handleSearchChange = useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
       if (event.target.value.length === 0 || event.target.value.length > 1) {
-        setSearchTerm(event.target.value);
+        setSearchTerm(event.target.value.toLowerCase());
+        setTimeout(() => inputRef.current?.focus(), 0);
```

Comment:
I like this fix. It's clever workaround.

I think we're using the menu list component for more than it was originally designed for. That's probably why we're running into the re-rendering issue. For context, here are the docs for the `Menu` component: https://v1.chakra-ui.com/docs/components/overlay/menu


None of those examples have same dynamic behavior that requires being focused on something.